### PR TITLE
Glossary: add 'aggregate' and modify the doc

### PR DESCRIPTION
**Pull request by Asu4ni**
related #340

  1. add item 'aggregate': differentiate aggregates from measures
  2. add item 'timespan'
  3. modify measures to aggregates if needed
  4. glossary minor fixes

### CODE REVIEW

File: doc/source/glossary.rst [hunk 1]
```diff
@@ -3,31 +3,42 @@ Glossary
...
 
    Measure
-     A datapoint tuple composed of timestamp and a value.
+     An incoming datapoint tuple sent to Gnocchi by the api. It is composed
+     of a timestamp and a value.
 
    Archive policy
-     A measure storage policy attached to a metric. It determines how long
-     measures will be kept in a metric and how they will be aggregated.
+     A aggregate storage policy attached to a metric. It determines how long
```
Comment (jd) [inline doc/source/glossary.rst]: An aggregate?

**Comment by Asu4ni** [comment]
I only changed the document parts. I don't understand why other parts failed.

**Comment by Asu4ni** [comment]
@chungg 
> can you leave the picture as is
I could, but why is it the case? Is there any problems with my image?

**Comment by Asu4ni** [comment]
@chungg, I remade the image not only because the text should be changed but also because the original one is less organized. Is it fine to use the image I made?

**Comment by jd** [comment]
I like the new image, and it's in SVG, so that's better. Which software did you use to build it?
I think it misses the link between coordinator and API (which barely use it, but still a bit). Maybe it was missing in the old schema too, but we should add it for completeness I think.

**Comment by chungg** [comment]
i don't have a problem with picture per se. it's just we need to know what tool was used or it'll be impossible to actually edit the picture later if you're not around :)

my only question is why the resolution is so small (or maybe it's just how github formats svg)... i have to change my browser to zoom in 300% to view it.

if i were to give feedback. i'd say drop the 'gnocchi' part of api and metricd. i'm pretty sure everyone knows we're talking about gnocchi

**Comment by jd** [comment]
> i don't have a problem with picture per se. it's just we need to know what tool was used or it'll be impossible to actually edit the picture later if you're not around :)

Yeah, my question. But it's SVG so it's vectorial and so actually editable with any software that knows SVG – so that's better than my crappy PNG screenshoted from Google Docs :p 

> my only question is why the resolution is so small (or maybe it's just how github formats svg)... i have to change my browser to zoom in 300% to view it.

Yes it's vectorial so you can zoom infinitely (DON'T TRY THOUGH) if you want, that's a github/browser issue.

> if i were to give feedback. i'd say drop the 'gnocchi' part of api and metricd. i'm pretty sure everyone knows we're talking about gnocchi

+1

**Comment by Asu4ni** [comment]
### Tool
It's Google Drawing. [Link](https://docs.google.com/drawings/d/1aHV86TPNFt7FlCLEjsTvV9FWoFYxXCaQOzfg7NdXVwM/edit?usp=sharing) I'm not sure how we will manage the share setting. Letting anyone with the link to edit it will be a mess.

### link between coordinator & api
Should it be a bi-direction arrow or a normal arrow (which side to point?) ? I don't know the relation between them. Please inform me of that.

**Comment by jd** [comment]
> I'm not sure how we will manage the share setting. Letting anyone with the link to edit it will be a mess.

Just read only share is OK. Add the link as a comment in the source of the rst so we know where to go to create a copy and edit.

> Should it be a bi-direction arrow or a normal arrow (which side to point?) ? I don't know the relation between them. Please inform me of that.

Same as metricd, bi directional I'd say.

**Comment by Asu4ni** [comment]
This link is for viewing: https://docs.google.com/drawings/d/1aHV86TPNFt7FlCLEjsTvV9FWoFYxXCaQOzfg7NdXVwM/edit?usp=sharing

And this link is for copying:
https://docs.google.com/drawings/d/1aHV86TPNFt7FlCLEjsTvV9FWoFYxXCaQOzfg7NdXVwM/copy

Which should I use? or both?
And what kind of format to put it in? In the 'alt' of the image?

**Comment by jd** [comment]
go with the view link

you can put comments in RST:
 http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#comments

**Comment by jd** [comment]
@Asu4ni it'd be good you squash all of those commits in one I think

File: doc/source/glossary.rst [hunk 2]
```diff
@@ -3,31 +3,42 @@ Glossary
...
-     Function used to aggregate multiple measures in one. For example, the
-     `min` aggregation method will aggregate the values of different measures
-     to the minimum value of all the measures in time range.
+     Function used to aggregate multiple measures into an aggregate. For
+     example, the `min` aggregation method will aggregate the values of
+     different measures to the minimum value of all the measures in the time
+     range.
+
+   Aggregate
+     An datapoint tuple generated from several measures according to the
```
Comment (chungg) [inline doc/source/glossary.rst]: s/An/A/

File: doc/source/nagios.rst [hunk 3]
```diff
@@ -10,7 +10,7 @@ service. It can be installed with pip::
 `Source`_ and `Documentation`_ are also available.
 
 Gnocchi-nagios collects perfdata files generated by `Nagios`_ or `Icinga`_;
-transforms them into Gnocchi resources, metrics and measures format; and
+transforms them into Gnocchi resources, metrics and aggregates format; and
```
Comment (chungg) [inline doc/source/nagios.rst]: i don't think nagios is passing in an aggregate.

**Pull request merged**